### ISSUE TITLE
[v2.1.1] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#110961]
### 🐛 Describe the bug

This issue is for tracking cherry-picks to the release branch. Following is [release branch](https://github.com/pytorch/pytorch/tree/release/2.1) for the 2.1.1 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 11/3): Cherry-pick post deadline (End of day 5PM PST)
* Phase 2 (after 11/3):  Perform extended integration/stability/performance testing based on Release Candidate builds.

**Only issues that have ‘cherry-picks’ in this tracker will be considered for the release.**

## Cherry-Pick Criteria

**Phase 1 (until 11/3):**

The Releng team relies on the cherry pick process to manage risk to release quality, i.e. by porting a small set of commit from trunk that are "must-have" into the release branch, we limit the change to the minimal to address pressing issues. Thus, not everything a developer land into the trunk will make it into the release. So, please consider the criteria below and follow the cherry picking process. Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent release (e.g. 2.1.0 for 2.1.1 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Low risk critical fixes for: silent correctness, backwards compatibility, crashes, deadlocks, (large) memory leaks
3. Fixes to new features being introduced in 2.1.0 release
4. Documentation improvements
5. Release branch specific changes (e.g. blocking ci fixes, change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @huydhn, @osalpekar, @malfet). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain. 

**Phase 2 (after 11/3):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.0).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/2.1
    git checkout release/2.1
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/2.1' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/2.1
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

See [HUD 2.1](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.1/1?per_page=50) 

### Versions

2.1.1

**Comment by tringwald** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/109326 

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/110969

Criteria Category:
* Documentation fixes

**Comment by soumith** [comment]
got request to include these two fixes from Mosaic:

* https://github.com/pytorch/pytorch/pull/109116

* https://github.com/pytorch/pytorch/pull/108915

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/107672

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/111039

Criteria Category:
* Required CI fix for linux Arm 64

**Comment by Skylion007** [comment]
A minor NCCL version dropped that has a bug fix. Should we open a PR onto 2.1 tree to update it?

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/111042

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111578

Criteria Category:
*  Critical CI fix for https://github.com/pytorch/pytorch/issues/100974

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110759

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111597

Criteria Category:
* Critical CI fix

**Comment by jataylo** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/111467

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111667

Criteria Category:
* Unit test fixes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111675

Criteria Category:
* CI Fixes

**Comment by wz337** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/109626

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111679

Criteria Category:
* Test for the critical fix(https://github.com/pytorch/pytorch/pull/108915) that is cherry-picked into Release2.1

**Comment by wz337** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/111096

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111687

Criteria Category:
* Critical Fix

**Comment by thiagocrepaldi** [comment]
Link to landed master PR (if applicable): 
* https://github.com/pytorch/pytorch/pull/110956

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111694

Criteria Category:
* Fix regression to TorchScript

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/111697

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/111801

Criteria Category:
* Critical CI fix

**Comment by tringwald** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/111703

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111814

Criteria Category:
* Memory Leak

**Comment by jcaip** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110420

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111845

Criteria Category:
* Critical Fixes

**Comment by eqy** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110665

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111875

Criteria Category:
* Critical fixes

**Comment by eqy** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110326

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111877

Criteria Category:
* Critical fixes

**Comment by BowenBao** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/111859

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111861

Criteria Category:
* Critical fixes: regression, crash

**Comment by malfet** [comment]
Link to the landed main commit:

- https://github.com/pytorch/builder/commit/c5e331c0858e37fedc047707466161dfe0cadff6

Lint to release branch PR:
- https://github.com/pytorch/builder/pull/1574

Criteria Category:
  - Regression from the previous release, crash

**Comment by huydhn** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110385

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/111942

Criteria Category:
* Critical CI fix

**Comment by Chillee** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110687

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111946

Criteria Category:
* Regression from previous release

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/111971

Criteria Category:
* Critical CI fix for docker release build

**Comment by malfet** [comment]
Link to landed trunk PR:
 - https://github.com/pytorch/pytorch/pull/111699

Lint to a release branch PR:
 - https://github.com/pytorch/pytorch/pull/111996

Criteria Category:
  - Fixes to regressions against the most recent release

**Comment by ev-br** [comment]
Link to landed trunk PR:
 - https://github.com/pytorch/pytorch/pull/110512

Lint to a release branch PR:
 -  https://github.com/pytorch/pytorch/pull/112011

Criteria Category:
  - Fixes to new features being introduced in 2.1.0 release

**Comment by ev-br** [comment]
Link to landed trunk PR:
  - trace frames with np.ndarray #110512
  - https://github.com/pytorch/pytorch/pull/107863

Lint to a release branch PR:
  - https://github.com/pytorch/pytorch/pull/112027
  
Criteria Category:
  - Fixes to new features being introduced in 2.1.0 release

This is an alternative to https://github.com/pytorch/pytorch/issues/110961#issuecomment-1778794463 : lump up two trunk PRs into a two separate commits of a backport PR. If wanted, can submit the backports separately, please advise.

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/111916

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/112026

Criteria Category:
*  Critical CI fix

---
@malfet merged

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/112046

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/112040

Criteria Category:
*  Critical CI docker release fix

---
@atalman merged

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/110575
* https://github.com/pytorch/pytorch/pull/110990

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/111835

Criteria Category:
*  Critical Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/111347

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/112163

Criteria Category:
*  Performance issue https://github.com/pytorch/pytorch/issues/110666

**Comment by kit1980** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/commit/d80174e2db679365f8b58ff8583bdc4af5a8b74c

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/112187

Criteria Category:
*  Low risk critical fixes for backwards compatibility,

**Comment by huydhn** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/111576

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/112265

Criteria Category:
*  Critical CI fix, GitHub MacOS runner

**Comment by malfet** [comment]
Link to the landed PR:
 - https://github.com/pytorch/pytorch/pull/109557

Link to the release branch PR:
 - https://github.com/pytorch/pytorch/pull/112432

Criteria Category:
  - Low risk critical fixes for  silent correctness on MPS->CPU copy

**Comment by huydhn** [comment]
Link to the landed PR:
 - https://github.com/pytorch/builder/pull/1575

Link to the release branch PR:
 - https://github.com/pytorch/builder/commit/77901321d9387eeb68ca9e2ddab7562883562cd1

Criteria Category:
  - Critical CI fixes

---
@huydhn merged

**Comment by wz337** [comment]
Got request from Mosaic to include this 1 fix.

Link to the landed PR:
* https://github.com/pytorch/pytorch/pull/112259

Link to the release branch PR:
* https://github.com/pytorch/pytorch/pull/112520

Criteria Category:

**Comment by huydhn** [comment]
Link to the landed PR:
 - N/A

Link to the release branch PR:
 - https://github.com/pytorch/pytorch/pull/112665

Criteria Category:
  - Critical CI fixes

**Comment by huydhn** [comment]
Link to the landed PR:
 - https://github.com/pytorch/builder/pull/1576

Link to the release branch PR:
 - https://github.com/pytorch/builder/pull/1577
 
Criteria Category:
  - Critical CI fixes.  This allows the build Docker images to be updated when cutting RC

**Comment by malfet** [comment]
Link to the landed trunk PR:
 - https://github.com/pytorch/pytorch/pull/112349

Link to the release branch PR:
- https://github.com/pytorch/pytorch/pull/112756

Criteria Category: 
- Fixes to regressions against the most recent release (didn't crash in 2.0.1, crashes in 2.1.0)

**Comment by malfet** [comment]
Link to the landed trunk PR:
 - https://github.com/pytorch/pytorch/pull/111672

Link to the release branch PR:
 - https://github.com/pytorch/pytorch/pull/112784

Criteria Category:
 - Fixes to regressions against the most recent release (didn't crash in 2.0.1, crashes in 2.1.0)

**Comment by huydhn** [comment]
Link to the landed trunk PR:
 - https://github.com/pytorch/pytorch/pull/112170

Link to the release branch PR:
 - https://github.com/pytorch/pytorch/pull/112790

Criteria Category:
 - Fixes to regressions against the most recent release https://github.com/pytorch/pytorch/issues/110941

**Comment by drisspg** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110893

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/112792

Criteria Category:
* Fixes to regression when trying to compile scaled dot product attention

**Comment by drisspg** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/112267

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/112795

Criteria Category:
* Silent correctness regression for scaled dot product attention

**Comment by drisspg** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/112673

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/112796

Criteria Category:
* Fixes a silent correctness issue with scaled dot product attention reported in: https://github.com/pytorch/pytorch/issues/112577

**Comment by malfet** [comment]
Link to landed trunk PR:
- https://github.com/pytorch/pytorch/pull/112995
- https://github.com/pytorch/pytorch/pull/112996

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/113096

Criteria Category:
- Fixes to new features being introduced in 2.1.0 release

**Comment by malfet** [comment]
Closing and unpinning, as final RC is out
